### PR TITLE
Add dashboard template refinements

### PR DESCRIPTION
**Pull request by danilo-leal**
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/material-ui/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

### CODE REVIEW

File: docs/data/material/getting-started/templates/dashboard/components/CustomizedTreeView.tsx [hunk 1]
```diff
@@ -182,24 +182,26 @@ const CustomTreeItem = React.forwardRef(function CustomTreeItem(
 
 export default function CustomizedTreeView() {
   return (
-    <Card variant="outlined">
-      <CardContent>
```
Comment (noraleonte) [inline docs/data/material/getting-started/templates/dashboard/components/CustomizedTreeView.tsx:186]: I think that eliminating the `CardContent` here messes with the components on the right side for the MD2 version 🤔 Is there any specific reason for this change?
image

File: docs/data/material/getting-started/templates/dashboard/components/HighlightedCard.tsx [hunk 2]
```diff
@@ -40,18 +37,22 @@ export default function HighlightedCard() {
...
+          Uncover performance and visitor insights with our data wizardry.
         </Typography>
       </CardContent>
-      <CardActions sx={{ justifyContent: 'flex-end' }}>
-        <Button size="small" color="secondary" endIcon={<ChevronRightRoundedIcon />}>
-          Get insights
-        </Button>
-      </CardActions>
+      <Button
+        variant="outlined"
```
Comment (noraleonte) [inline docs/data/material/getting-started/templates/dashboard/components/HighlightedCard.tsx]: I wonder if we can/should tweak this outlined version a bit 🤔 I feel like it blends in with the card background and I personally find it a bit confusing
image

It's also a bit weird on the MD2 version without the `CardAction`
image

File: docs/data/material/getting-started/templates/dashboard/components/MenuButton.tsx [hunk 3]
```diff
@@ -15,9 +15,17 @@ export default function MenuButton({
...
-      sx={{ '& .MuiBadge-badge': { right: 2, top: 2 } }}
+      sx={{
+        '& .MuiBadge-badge': { right: 2, top: 2 },
+      }}
     >
-      <IconButton size="small" {...props} />
+      <IconButton
+        size="small"
+        sx={{
+          '& .MuiSvgIcon-root': { width: '1.125rem', height: '1.125rem' },
```
Comment (noraleonte) [inline docs/data/material/getting-started/templates/dashboard/components/MenuButton.tsx]: We should probably add these to the theme instead. In MD2 they feel too small IMO
image

File: docs/data/material/getting-started/templates/dashboard/getDashboardTheme.tsx [hunk 4]
```diff
@@ -429,25 +429,21 @@ export default function getDashboardTheme(mode: PaletteMode): ThemeOptions {
             textTransform: 'none',
             fontWeight: theme.typography.fontWeightMedium,
             letterSpacing: 0,
-            color: gray[600],
-            border: '1px solid',
-            borderColor: gray[200],
+            color: theme.palette.text.primary,
+            border: `1px solid ${theme.palette.divider}`,
             backgroundColor: gray[50],
```
Comment (noraleonte) [inline docs/data/material/getting-started/templates/dashboard/getDashboardTheme.tsx:434]:
```suggestion
            backgroundColor: gray[50],
            '& .MuiSvgIcon-root': { width: '1.125rem', height: '1.125rem' },
```

**Review by noraleonte — APPROVED** [review]
Thanks for going through this 🎉 I left a few comments for some of the changes, but once those are discussed I think we can merge this 🎉

File: docs/data/material/getting-started/templates/dashboard/components/CustomizedTreeView.tsx (hunk 1 above)
Comment (danilo-leal) [reply docs/data/material/getting-started/templates/dashboard/components/CustomizedTreeView.tsx:186]: Oh, great call, I hadn't toggled back to the default theme at all 😅

File: docs/data/material/getting-started/templates/dashboard/components/HighlightedCard.tsx (hunk 2 above)
Comment (danilo-leal) [reply docs/data/material/getting-started/templates/dashboard/components/HighlightedCard.tsx]: Yeah, maybe it's too little affordance!

File: docs/data/material/getting-started/templates/dashboard/components/MenuButton.tsx (hunk 3 above)
Comment (danilo-leal) [reply docs/data/material/getting-started/templates/dashboard/components/MenuButton.tsx]: Great call!

**Review by zanivan — APPROVED** [review]
Thanks @danilo-leal for such great improvements overall! 

—-

One thing, though, I think we should decide upon which chart we're going to use, bars or lines (or maybe both, but changing title/legend labels?) What do you think @noraleonte @danilo-leal ?

Screenshot 2024-05-28 at 11 55 39

**Comment by noraleonte** [comment]
@zanivan Tbh, I'm fine with either 🙈 Jun suggested line charts on twitter so we can go with that

**Pull request merged**